python/lsst/daf/butler/core/quantum.py

```diff
@@ -77,7 +77,7 @@ def _reconstructDatasetRef(
 class SerializedQuantum(BaseModel):
     """Simplified model of a `Quantum` suitable for serialization."""
 
-    taskName: str
+    taskName: str | None
```

Comment:
I'm a little worried that something wants to set this to `None`, but that's probably also something to punt.

Reply:
The error is:

```
python/lsst/daf/butler/core/quantum.py:380: error: Argument "taskName" to "SerializedQuantum" has incompatible type "Optional[str]"; expected "str"  [arg-type]
```

and it's because `Quantum` is allowed to accept an optional task name.